infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -506,7 +528,17 @@ public void prepareSiteRoleChange() {
      */
     @Override
     public int queryReplicationStatus() {
-        return replicationManager.queryReplicationStatus();
+        if (localClusterDescriptor.getRole() == ClusterRole.ACTIVE) {
+            // If not leader, replication manager might be null
+            if (replicationManager != null) {
+                return replicationManager.queryReplicationStatus();
+            }
+            return 0;
```

Comment:
can we add some logging warning, before this return 0. Cause then we don't know if it means that 0 is pending as it all was transferred or the replication never started so 0 is pending.